src/pages/Scheduling/ScheduleExceptions.tsx

```diff
@@ -112,15 +126,38 @@ const ScheduleExceptionItem = (
             </span>
           </div>
         </div>
-        <Button
-          variant="secondary"
-          size="sm"
-          disabled={isPending}
-          onClick={() => deleteException(undefined)}
-        >
-          <CareIcon icon="l-minus-circle" className="text-base" />
-          <span className="ml-2">{t("remove")}</span>
-        </Button>
+        <AlertDialog open={open} onOpenChange={setOpen}>
+          <AlertDialogTrigger asChild>
+            <Button variant="secondary" size="sm" disabled={isPending}>
+              <CareIcon icon="l-minus-circle" className="text-base" />
+              <span className="ml-2">{t("remove")}</span>
+            </Button>
+          </AlertDialogTrigger>
+          <AlertDialogContent>
+            <AlertDialogHeader>
+              <AlertDialogTitle>{t("are_you_sure")}</AlertDialogTitle>
+              <AlertDialogDescription>
+                <Alert variant="destructive" className="mt-4">
+                  <AlertTitle>{t("warning")}</AlertTitle>
+                  <AlertDescription>
+                    {t("this_will_permanently_remove_the_template")}
+                  </AlertDescription>
+                </Alert>
+              </AlertDialogDescription>
+            </AlertDialogHeader>
+            <AlertDialogFooter>
+              <AlertDialogCancel>{t("cancel")}</AlertDialogCancel>
+              <AlertDialogAction
+                onClick={() => {
+                  deleteException(undefined);
```

Comment:
update API definition's `TBody` to be of type `void` to fix this

```suggestion
                  deleteException();
```

Reply:
updated